File: WS_EUREKABANK_SOAPDOTNET_GR04/02.CLICON/EB_DOTNET_SOAP_CliCon/ec.edu.monster.vista/CliCon_Vista.cs
using EB_DOTNET_SOAP_CliCon.ec.edu.monster.controlador;
using EB_DOTNET_SOAP_CliCon.ec.edu.monster.vista;
using System;

namespace EB_DOTNET_SOAP_CliCon
{
    public class CliCon_Vista
    {
        static void Main(string[] args)
        {
            var scanner = new ConsoleInputHelper();
            var controlador = new CliCon_Controlador();

            Console.WriteLine("\n=== E U R E K A B A N K ===\n");

            Console.Write("Usuario: ");
            string usuario = Console.ReadLine();
            Console.Write("Password: ");
            string password = Console.ReadLine();

            var login = controlador.ValidarIngreso(usuario, password);
            if (!string.Equals(login, "Exitoso", StringComparison.OrdinalIgnoreCase))
            {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine("\nAcceso denegado");
                Console.ResetColor();
                return;
            }

            Console.ForegroundColor = ConsoleColor.Green;
            Console.WriteLine("\nAcceso exitoso");
            Console.ResetColor();

            int opcion;
            do
            {
                Console.WriteLine("\n1. Consultar movimientos");
                Console.WriteLine("2. Realizar depósito");
                Console.WriteLine("3. Realizar retiro");
                Console.WriteLine("4. Realizar transferencia");
                Console.WriteLine("5. Salir");
                Console.Write("\nSeleccione: ");
                opcion = scanner.ReadInt();

                switch (opcion)
                {
                    case 1:
                        Console.Write("\nCuenta: ");
                        var cuenta = Console.ReadLine();
                        var movimientos = controlador.ObtenerMovimientos(cuenta);
                        if (movimientos.Count == 0) Console.WriteLine("Sin movimientos.");
                        else MovimientoView.MostrarMovimientos(movimientos);
                        break;

                    case 2:
                        Console.Write("\nCuenta depósito: ");
                        var cDep = Console.ReadLine();
                        Console.Write("Importe: ");
                        var impDep = scanner.ReadDecimal();
                        var dep = controlador.RegistrarDeposito(cDep, impDep);
                        Console.WriteLine(dep.estado == 1
                            ? $"Depósito OK. Saldo actual: {dep.saldo:N2}"
                            : "Depósito falló.");
                        break;

                    case 3:
                        Console.Write("\nCuenta retiro: ");
                        var cRet = Console.ReadLine();
                        Console.Write("Importe: ");
                        var impRet = scanner.ReadDecimal();
                        var ret = controlador.RegistrarRetiro(cRet, impRet);
                        Console.WriteLine(ret.estado == 1
                            ? $"Retiro OK. Saldo actual: {ret.saldo:N2}"
                            : "Retiro falló.");
                        break;

                    case 4:
                        Console.Write("\nCuenta origen: ");
                        var cOri = Console.ReadLine();
                        Console.Write("Cuenta destino: ");
                        var cDes = Console.ReadLine();
                        Console.Write("Importe: ");
                        var impTr = scanner.ReadDecimal();
                        var tr = controlador.RegistrarTransferencia(cOri, cDes, impTr);
                        Console.WriteLine(tr.estado == 1
                            ? $"Transferencia OK. Saldo origen: {tr.saldo:N2}"
                            : "Transferencia falló.");
                        break;

                    case 5:
                        Console.WriteLine("Saliendo...");
                        break;

                    default:
                        Console.WriteLine("Opción inválida.");
                        break;
                }
            }
            while (opcion != 5);
        }
    }


    public class ConsoleInputHelper
    {
        public int ReadInt()
        {
            while (true)
            {
                if (int.TryParse(Console.ReadLine(), out var v)) return v;
                Console.Write("Número inválido. Intente: ");
            }
        }

        public decimal ReadDecimal()
        {
            while (true)
            {
                if (decimal.TryParse(Console.ReadLine(),
                    System.Globalization.NumberStyles.Number,
                    System.Globalization.CultureInfo.InvariantCulture,
                    out var v)) return v;
                Console.Write("Número inválido. Intente: ");
            }
        }
    }

}

File: WS_EUREKABANK_RESTFULLJAVA_GR04/02.CLICON/EB_Restfull_JAVA_CliCon/src/ec/edu/monster/vista/CliCon_Vista.java
package ec.edu.monster.vista;

import ec.edu.monster.controlador.CliCon_Controlador;
import java.util.Scanner;

public class CliCon_Vista {

    public static void main(String[] args) {

        CliCon_Controlador controlador = new CliCon_Controlador();
        Scanner scanner = new Scanner(System.in);

        int opcion;

        // ASCII ART (lo dejo igual que tú)
        System.out.println(" ███╗    ███╗ ██████╗ ███╗   ██╗██████╗███████╗███████╗██████╗ ");
        System.out.println(" ████╗ ████║██╔═══ ██╗████╗  ██║██╔═════╝ ╚══ ██╔══╝ ██╔═══════╝ ██╔══██╗ ");
        System.out.println(" ██╔████╔██║██║    ██║██╔██╗ ██║  ████╗     ██║   ████╗     █████╔╝ ");
        System.out.println(" ██║╚██╔╝ ██║██║    ██║██║╚██╗██║      ██║    ██║    ██╔══╝     ██╔══██║ ");
        System.out.println(" ██║ ╚═╝   ██║╚██████╔╝██║ ╚████║██████║    ██║    ███████╗██║   ██║ ");
        System.out.println(" ╚══╝       ╚══╝ ╚═════════╝ ╚══╝   ╚═════╝╚═════════╝   ╚═══╝    ╚═══════════╝╚══╝   ╚═══╝   ");

        System.out.println("\033[34m\n--------------------\033[0m");
        System.out.println("\033[34m     BIENVENIDO     \033[0m");
        System.out.println("\033[34m--------------------\033[0m");

        // === Login contra el servidor REST ===
        System.out.print("\nIngrese su usuario: ");
        String usuario = scanner.nextLine();
        System.out.print("Ingrese su contraseña: ");
        String password = scanner.nextLine();

        boolean acceso = controlador.login(usuario, password);

        if (acceso) {
            System.out.println("\033[32m\nAcceso exitoso\033[0m");

            do {
                System.out.println("\033[34m\n==========================\033[0m");
                System.out.println("\033[34m       MENÚ PRINCIPAL       \033[0m");
                System.out.println("\033[34m==========================\033[0m\n");
                System.out.println("1. Consultar movimientos");
                System.out.println("2. Realizar depósito");
                System.out.println("3. Realizar retiro");
                System.out.println("4. Realizar transferencia");
                System.out.println("5. Salir");
                System.out.print("\033[34m\nSeleccione una opción: \033[0m");

                while (!scanner.hasNextInt()) {
                    System.out.println("\033[31mOpción no válida. Ingrese un número.\033[0m");
                    scanner.next(); 
                }
                opcion = scanner.nextInt();
                scanner.nextLine(); // limpiar buffer

                switch (opcion) {
                    case 1:
                        System.out.println("\033[35m\n============================\033[0m");
                        System.out.println("\033[35m         MOVIMIENTOS        \033[0m");
                        System.out.println("\033[35m============================\033[0m");
                        System.out.print("\n\033[35mIngrese el número de cuenta: \033[0m");
                        String cuentaConsulta = scanner.nextLine();

                        String movimientos = controlador.obtenerMovimientos(cuentaConsulta);
                        System.out.println();
                        System.out.println(movimientos);
                        break;

                    case 2:
                        System.out.println("\033[36m\n============================\033[0m");
                        System.out.println("\033[36m         DEPÓSITO          \033[0m");
                        System.out.println("\033[36m============================\033[0m");
                        System.out.print("\033[36m\nIngrese el número de cuenta: \033[0m");
                        String cuentaDeposito = scanner.nextLine();
                        System.out.print("\033[36mIngrese el importe a depositar: \033[0m");
                        double importe = scanner.nextDouble();
                        scanner.nextLine();

                        String resultadoDeposito = controlador.registrarDeposito(cuentaDeposito, importe);
                        System.out.println("\n" + resultadoDeposito);
                        break;

                    case 3:
                        System.out.println("\033[35m\n============================\033[0m");
                        System.out.println("\033[35m         RETIRO            \033[0m");
                        System.out.println("\033[35m============================\033[0m");
                        System.out.print("\033[35m\nIngrese el número de cuenta: \033[0m");
                        String cuentaRetiro = scanner.nextLine();
                        System.out.print("\033[35mIngrese el importe a retirar: \033[0m");
                        double importeRetiro = scanner.nextDouble();
                        scanner.nextLine();

                        String resultadoRetiro = controlador.registrarRetiro(cuentaRetiro, importeRetiro);
                        System.out.println("\n" + resultadoRetiro);
                        break;

                    case 4:
                        System.out.println("\033[36m\n============================\033[0m");
                        System.out.println("\033[36m       TRANSFERENCIA        \033[0m");
                        System.out.println("\033[36m============================\033[0m");

                        System.out.print("\033[36m\nIngrese el número de cuenta origen: \033[0m");
                        String cuentaOrigen = scanner.nextLine();
                        System.out.print("\033[36mIngrese el número de cuenta destino: \033[0m");
                        String cuentaDestino = scanner.nextLine();
                        System.out.print("\033[36mIngrese el importe a transferir: \033[0m");
                        double importeTransferencia = scanner.nextDouble();
                        scanner.nextLine();

                        String resultadoTransferencia = controlador.registrarTransferencia(cuentaOrigen, cuentaDestino, importeTransferencia);
                        System.out.println("\n" + resultadoTransferencia);
                        break;

                    case 5:
                        System.out.println("\nSaliendo del sistema. ¡Gracias!");
                        break;

                    default:
                        System.out.println("\033[31mOpción no válida. Intente nuevamente.\033[0m");
                }

                System.out.println();
            } while (opcion != 5);
        } else {
            System.out.println("\033[31mAcceso denegado\033[0m");
        }

        scanner.close();
    }
}
